### PR TITLE
Added example project, basic tests, and CodeCov

### LINKED ISSUE #28: Create example application
- [x] Move sources files into `Sources` folder
- [x] Create example project inside xcodeproj, including a mp4 video asset
- [x] Add tests using mp4 video asset
- [x] Change podspec to only target source files in `Sources` folder
- [x] Investigate changes necessary to Carthage

### LINKED ISSUE #49: Add Codecov

### PR DESCRIPTION
**Pull request by dingwilson**
Closes #28 
Closes #49 

**Make sure to squash and merge the commits**

### Tasks:
- [x] Add example project
- [x] Move source files to `Sources/` folder
- [x] Ensure jazzy docs don't pick up files from example project
- [ ] Ensure this doesn't break Cocoapods
- [ ] Ensure this doesn't break Carthage
- [x] Add tests
- [x] Add Codecov Integration (#49)

### CODE REVIEW

**Comment by dingwilson** [comment]
@quanvo87 this is a big PR. I've tested things as much as I can, and I _think_ most everything works. The only things I'm unsure about are the new layout breaking CocoaPods/Carthage. Unless there is some way you can think of to test these, I think we're ready to merge.

File: SwiftVideoBackground/Sources/VideoBackgroundError.swift [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import Foundation
```
Comment (quanvo87) [inline SwiftVideoBackground/Sources/VideoBackgroundError.swift:1]: No idea how this was working before without this import lol

**Review by quanvo87 — APPROVED** [review]
BEAST

**Comment by quanvo87** [comment]
Hm would that cocoapods linting thing test it?

**Comment by quanvo87** [comment]
`pod spec lint`/`pod lib lint`

File: SwiftVideoBackground/Sources/VideoBackgroundError.swift (hunk 1 above)
Comment (dingwilson) [reply SwiftVideoBackground/Sources/VideoBackgroundError.swift:1]: 😂😂

**Comment by dingwilson** [comment]
Good idea...

`pod spec lint` errors out due to tag 2.1.1 not existing on GH - nothing much I can do about that

`pod lib lint` gives a success message

**Pull request merged**